src/App.jsx

```diff
@@ -4,21 +4,28 @@ import AddTodoForm from './AddTodoForm.jsx'
 import { useState, useEffect } from 'react'
 import React from 'react'
 
-const useSemiPersistentState = () => {
-  const [todoList, setTodoList] = useState(()=>{
-    const saved = localStorage.getItem('savedTodoList')
-    return saved ? JSON.parse(saved) : [];
-  });
+function App() {
+  const [todoList, setTodoList] = useState([])
+  const [isLoading, setIsLoading] = useState(true)
 
-  useEffect(()=>{
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList))
-  },[todoList]);
 
-  return [todoList, setTodoList]
-}
+  useEffect(()=>{
+    const fetchData = new Promise((resolve)=>{
+      setTimeout(()=>{
+        resolve({data:{todoList:JSON.parse(localStorage.getItem('savedTodoList'))||[]}})
+      },2000)
+    })
+    fetchData.then(result =>{
+      setTodoList(result.data.todoList)
+      setIsLoading(false)
+    })
```

Comment:
a `catch` clause could be useful down the line, especially once real network requests start happening. fine for now